internal/graph/singleflight_resolver.go

```diff
@@ -90,3 +98,14 @@ func copyResolveResponse(original ResolveCheckResponse) ResolveCheckResponse {
 		},
 	}
 }
+
+func singleflightRequestStateAttribute(isShared bool, isUnique bool) attribute.KeyValue {
```

Comment:
I separated this function out to not convolute the inside of the resolver itself; it became messy with the nested if/elses.